src/server/scripts/Shadowlands/SepulcherOfTheFirstOnes/boss_anduin_wrynn.cpp

```diff
@@ -0,0 +1,4307 @@
+        };
+        forceAttack(DATA_SYLVANAS_WINDRUNNER_ANDUIN);
+        forceAttack(DATA_UTHER_THE_LIGHTBRINGER_ANDUIN);
+        forceAttack(DATA_JAINA_PROUDMOORE_ANDUIN);
+    }
+
+    void MoveKnightsOnPlatform()
+    {
+        if (Creature* rahm = instance->GetCreature(DATA_QUARTERMASTER_RAHM_ANDUIN))
+            rahm->NearTeleportTo(QuartermasterRahmPos, false);
+
+        std::list<Creature*> knights;
+        me->GetCreatureListWithOptionsInGrid(knights, 150.0f, { .CreatureId = NPC_KNIGHT_OF_EBON_BLADE_ANDUIN });
+
+        for (Creature* knight : knights)
+        {
+            if (knight->HasStringId("left_knight"))
+                knight->NearTeleportTo(leftKnightPos, false);
+            else
+                knight->NearTeleportTo(rightKnightPos, false);
+        }
+    }
+
+    void ClearDebuffs()
+    {
+        instance->DoRemoveAurasDueToSpellOnPlayers(SPELL_HOPEBREAKER_DEBUFF);
+        instance->DoRemoveAurasDueToSpellOnPlayers(SPELL_BLASPHEMY_OVERCONFIDENCE);
+        instance->DoRemoveAurasDueToSpellOnPlayers(SPELL_BLASPHEMY_HOPELESSNESS);
+        instance->DoRemoveAurasDueToSpellOnPlayers(SPELL_DOMINATION_WORD_PAIN);
+        instance->DoRemoveAurasDueToSpellOnPlayers(SPELL_BEFOULED_BARRIER_DEBUFF);
+        instance->DoRemoveAurasDueToSpellOnPlayers(SPELL_BEFOULED_BARRIER_EXPLODE);
+        instance->DoRemoveAurasDueToSpellOnPlayers(SPELL_LOST_SOUL_DIMENSION);
+        instance->DoRemoveAurasDueToSpellOnPlayers(SPELL_FORCE_OF_WILL);
+        instance->DoRemoveAurasDueToSpellOnPlayers(SPELL_SCARRED_SOUL);
+        instance->DoRemoveAurasDueToSpellOnPlayers(SPELL_WEATHER_COSMETIC);
+        instance->DoRemoveAurasDueToSpellOnPlayers(SPELL_REMORSELESS_WINTER_PERIODIC);
+        instance->DoRemoveAurasDueToSpellOnPlayers(SPELL_HOPELESSNESS_HOPELESSNESS_AREATRIGGER);
+    }
+
+    void JustAppeared() override
+    {
+        scheduler.ClearValidator();
+        HandleIntroduction();
+        me->SetEmoteState(EMOTE_STATE_READY2H);
+        me->ModifyPower(me->GetPowerType(), 0);
+        DoCastSelf(SPELL_ANDUIN_PROGRESSION_AURA);
+        DoCastSelf(SPELL_WILLPOWER_DISPLAY, true);
+        DoCastSelf(SPELL_DARK_ZEAL_AURA, true);
+        instance->DoUpdateWorldState(WORLD_STATE_ANDUIN_ENCOUNTER_STARTED, 0);
+        instance->DoUpdateWorldState(WORLD_STATE_ANDUIN_INTERMISSION, 0);
+    }
+
+    void EnterEvadeMode(EvadeReason /*why*/) override
+    {
+        ClearDebuffs();
+        Talk(SAY_DISENGAGE);
+        if (_encounterEnded == true)
+            return;
+
+        _DespawnAtEvade();
+        summons.DespawnAll();
+        instance->DoUpdateWorldState(WORLD_STATE_ANDUIN_ENCOUNTER_STARTED, 0);
+        instance->SendEncounterUnit(ENCOUNTER_FRAME_DISENGAGE, me);
+    }
+
+    void Reset() override
+    {
+        _Reset();
+        _dominationWordCount = 0;
+        me->SummonCreatureGroup(SPAWN_GROUP_INITIAL);
+        events.Reset();
+        scheduler.CancelAll();
+        me->SetPower(POWER_ENERGY, 0);
```

Comment:
me->SetPower(me->GetPowerType(), 0);

Reply:
doesnt matter, as we only have energy anyway